src/performance.js

```diff
@@ -0,0 +1,184 @@
+    if (_worstTicks.length === 0) { return 0; }
+
+    var worstTicksAverage = 0;
+
+    for (var i = 0; i < _worstTicks.length; i++) {
+      worstTicksAverage += _worstTicks[i];
+    }
+
+    return worstTicksAverage /= _worstTicks.length;
+  };
+
+  function onTick (dt) {
+    if (!_canMeasure) { return; }
+
+    var now = Date.now();
+    var delta = now - _lastTick;
+    var isWorstDelta = (_worstTicks.length > 0 && delta >= _worstTicks[0]);
+    
+    _lastTick = now;
+    
+    if (isWorstDelta || _worstTicks.length === 0) {
+      _addWorstTick(delta);
+    } else {
+      _ticksSinceLastWorstUpdate++;
+    }
+
+    if (_ticksSinceLastWorstUpdate > TICKS_TIL_INCREASE_SCORE) {
+      _worstTicks.pop();
+      _ticksSinceLastWorstUpdate = 0;
+    }
+  };
+
+  /**
+   * @method clear - clears all the information saved by past measures
+   */
+  this.clear = function () {
+    _worstTicks = [];
+    _ticksSinceLastWorstUpdate = 0;
+  };
+
+  /**
+   * @method pause - pause the score measurement
+   */
+  this.pause = function () {
+    _canMeasure = false;
+  };
+
+  /**
+   * @method resume - resume the score measurement
+   */
+  this.resume = function () {
+    _lastTick = Date.now();
+    _canMeasure = true;
+  };
+
+  /**
+   * @method setTargetFPSRange - sets the range of FPS used to calculate the score 
+   *   @arg {number} min - minimum value with a score of 0
+   *   @arg {number} max - maximum value with a score of 100
+   */
+  this.setTargetFPSRange = function (min, max) {
+    _minFPS = min;
+    _maxFPS = max;
+  };
+
+  /**
+   * @method getPerformanceScore - sets the range of FPS used to calculate the score 
+   */
+  this.getPerformanceScore = function () {
+    var worstTicksAverage = _getWorstTicksAverage();
+    var ticksPerSecond = 1000 / worstTicksAverage;
+    var adjustedTicksPerSecond = Math.min(ticksPerSecond, 60);
+    var mappedScore = _mapFPSToPerformanceScore(adjustedTicksPerSecond);
+
+    return Math.max(0, mappedScore);
+  };
+
+  /**
+   * @method getAdjustedParticleCount - returns a number of particles depending 
+   *   on the current performance score to improve performance
+   * @arg {number} count - number of particles that are going to be adjusted
+   * @arg {number} performanceScore - threshold for the calculations
+   * @arg {boolean} allowReduction - boolean used to define if the particles are 
+   *   going to be reduced or totally disabled
+   * @returns {number} count - number of particles 
+   */
+  this.getAdjustedParticleCount = function (count, performanceScore, allowReduction) {
+    var currCount = count;
+    var mR = this.getPerformanceScore();
+    var pR = performanceScore || DEFAULT_RANK;
+    var aR = (typeof allowReduction !== 'undefined')
+      ? allowReduction
+      : DEFAULT_ALLOW_REDUCTION;
+
+    if (mR < pR) {
+      currCount = (aR)
+        ? count * mR / pR
+        : 0;  
+    } 
+
+    return currCount;
+  }; 
+
+  this.init = function () {
```

Comment:
`init` should always be the first function in the Class (after private variable declaration) - this is to be consistent with the rest of our Classes